### PR TITLE
Warn for #2615 - duplicate insertion in same bulk

### PR DESCRIPTION
**Pull request by dnil**
This PR fixes a missing warning when duplicate variant_id occur in same insertion bulk.

The actual issue with SVs with same start and nondescript ALT getting the same variant_id is left open. Changing the id structure for the entire db would mean an epic major change. We could "cheat" a little with like tweakin the start a tiny bit, which may not mean so much for a large SV, but we are getting into a territory where we really need to sit back and think about the best to do this.

**How to test**:
1. See test data in issue #2615

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

Screenshot 2021-05-25 at 09 22 26

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

File: scout/adapter/mongo/variant_loader.py [hunk 1]
```diff
@@ -312,12 +312,11 @@ def upsert_variant(self, variant_obj):
         try:
             result = self.variant_collection.insert_one(variant_obj)
         except DuplicateKeyError as err:
-            LOG.debug("Variant %s already exists in database", variant_obj["_id"])
+            LOG.warning("Variant %s already exists in database - modifying", variant_obj["_id"])
```
Comment (northwestwitch) [inline scout/adapter/mongo/variant_loader.py:315]: 👍🏻

File: scout/adapter/mongo/variant_loader.py [hunk 2]
```diff
@@ -338,6 +337,7 @@ def load_variant_bulk(self, variants):
         except (DuplicateKeyError, BulkWriteError) as err:
             # If the bulk write is wrong there are probably some variants already existing
             # In the database. So insert each variant
+            LOG.debug("Bulk insertion failed - attempting separate upsert")
```
Comment (northwestwitch) [inline scout/adapter/mongo/variant_loader.py:340]: Why debug and not warning here?

File: scout/adapter/mongo/variant_loader.py [hunk 3]
```diff
@@ -312,12 +312,11 @@ def upsert_variant(self, variant_obj):
         try:
             result = self.variant_collection.insert_one(variant_obj)
         except DuplicateKeyError as err:
-            LOG.debug("Variant %s already exists in database", variant_obj["_id"])
+            LOG.warning("Variant %s already exists in database - modifying", variant_obj["_id"])
             result = self.variant_collection.find_one_and_update(
                 {"_id": variant_obj["_id"]},
                 {"$set": {"compounds": variant_obj.get("compounds", [])}},
             )
-            variant = self.variant_collection.find_one({"_id": variant_obj["_id"]})
```
Comment (northwestwitch) [inline scout/adapter/mongo/variant_loader.py:320]: This wasn't used?

File: scout/adapter/mongo/variant_loader.py [hunk 4]
```diff
@@ -488,7 +488,17 @@ def _load_variants(
                     bulk = {}
 
                 current_region = new_region
-                bulk[var_id] = variant_obj
+                if var_id in bulk:
```
Comment (northwestwitch) [inline scout/adapter/mongo/variant_loader.py:491]: Perhaps a stupid question but I feel I have a limited understanding the bulks: what if the variant is in another bulk instead?

File: scout/adapter/mongo/variant_loader.py [hunk 5]
```diff
@@ -488,7 +488,17 @@ def _load_variants(
...
+                    LOG.warning(
```
Comment (northwestwitch) [inline scout/adapter/mongo/variant_loader.py:492]: 👍🏻

**Review by northwestwitch — APPROVED** [review]
The code looks good, and I see that it's tested! 👍🏻

File: scout/adapter/mongo/variant_loader.py (hunk 4 above)
Comment (dnil) [reply scout/adapter/mongo/variant_loader.py:491]: Then the warning that was already in the code should work (database throws an exception, and each variant in the bulk is upserted separately to get to the issue and ensure that as many as possible have been inserted.

File: scout/adapter/mongo/variant_loader.py (hunk 3 above)
Comment (dnil) [reply scout/adapter/mongo/variant_loader.py:320]: Apparently not.. Timesaving! 😁

File: scout/adapter/mongo/variant_loader.py (hunk 2 above)
Comment (dnil) [reply scout/adapter/mongo/variant_loader.py:340]: Good point - I actually ran the "test"with debug. Will fix!

**Pull request merged**